plugins/coordinator/plugin.go

```diff
@@ -0,0 +1,153 @@
+	ErrEnvironmentVariableInvalidSeed   = errors.New("invalid coordinator seed. check environment variable COO_SEED")
+)
+
+func LoadSeedFromEnvironment() (trinary.Hash, error) {
+	viper.BindEnv("COO_SEED")
+	seed := viper.GetString("COO_SEED")
+
+	if len(seed) == 0 {
+		return "", ErrEnvironmentVariableSeedNotSet
+	}
+
+	if !guards.IsTransactionHash(seed) {
+		return "", ErrEnvironmentVariableInvalidSeed
+	}
+
+	return seed, nil
+}
+
+func configure(plugin *node.Plugin) {
+
+	log = logger.NewLogger(plugin.Name)
+
+	// set the node as synced at startup, so the coo plugin can select tips
+	tanglePlugin.SetUpdateSyncedAtStartup(true)
+
+	var err error
+	seed, err = LoadSeedFromEnvironment()
+	if err != nil {
+		log.Panic(err)
+	}
+
+	cooAddress := config.NodeConfig.GetString(config.CfgCoordinatorAddress)
+	securityLvl = config.NodeConfig.GetInt(config.CfgCoordinatorSecurityLevel)
+	merkleTreeDepth = config.NodeConfig.GetInt(config.CfgCoordinatorMerkleTreeDepth)
+	mwm = config.NodeConfig.GetInt(config.CfgCoordinatorMWM)
+	stateFilePath = config.NodeConfig.GetString(config.CfgCoordinatorStateFilePath)
+	merkleTreeFilePath = config.NodeConfig.GetString(config.CfgCoordinatorMerkleTreeFilePath)
+	milestoneIntervalSec = config.NodeConfig.GetInt(config.CfgCoordinatorIntervalSeconds)
+	checkpointTransactions = config.NodeConfig.GetInt(config.CfgCoordinatorCheckpointTransactions)
+
+	if _, err = os.Stat(merkleTreeFilePath); os.IsNotExist(err) {
+		log.Panic(errors.Wrapf(ErrMerkleTreeFileNotFound, "%v", merkleTreeFilePath))
+	}
+
+	coordinatorMerkleTree, err = loadMerkleTreeFile(merkleTreeFilePath)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	if cooAddress != coordinatorMerkleTree.Root {
+		log.Panic(errors.Wrapf(ErrCooAddressDoesNotMatchMerkleTree, "%v != %v", cooAddress, coordinatorMerkleTree.Root))
+	}
+
+	_, err = os.Stat(stateFilePath)
+	stateFileExists := !os.IsNotExist(err)
+
+	if *bootstrap {
+		if stateFileExists {
+			log.Panic(ErrNetworkBootstrapped)
+		}
+
+		coordinatorState = &CoordinatorState{}
+		coordinatorState.latestMilestoneHash = consts.NullHashTrytes
+		coordinatorState.latestMilestoneIndex = milestone.Index(*startIndex)
+		if coordinatorState.latestMilestoneIndex == 0 {
+			// start with milestone 1 at least
+			coordinatorState.latestMilestoneIndex = 1
+		}
+		coordinatorState.latestMilestoneTime = 0
+		coordinatorState.latestMilestoneTransactions = []trinary.Hash{consts.NullHashTrytes}
+		bootstrapped = false
+	} else {
+		if !stateFileExists {
+			log.Panic(ErrStateFileNotFound)
+		}
+
+		coordinatorState, err = loadStateFile(stateFilePath)
+		if err != nil {
+			log.Panic(err)
+		}
+		bootstrapped = true
+	}
+}
+
+func run(plugin *node.Plugin) {
+
+	// create a background worker that issues milestones
+	daemon.BackgroundWorker("Coordinator", func(shutdownSignal <-chan struct{}) {
+		timeutil.Ticker(issueNextCheckpointOrMilestone, time.Second*time.Duration(milestoneIntervalSec)/time.Duration(checkpointTransactions), shutdownSignal)
```

Comment:
This will have to be replaced by calling the corresponding function on the Coordinator struct which will be created instead of calling sole functions.